### PR TITLE
58531 Remove BDD SHA alerts from non BDD views

### LINKED ISSUE #58531: Remove BDD SHA alerts from non BDD views
## Issue Description
An issue was found while regression testing. 

**Steps to reproduce**
1. Start a new claim
2. Under 'Are you on active duty right now?' select No. 
3. Continue the 526 form until you get to the evidence types page `/disability/file-disability-claim-form-21-526ez/supporting-evidence/additional-evidence`
4. Under 'Is there any evidence you'd like us to review as part of your claim?' select 'Yes'
5. Under 'What type of evidence do you want us to review as part of your claim?" select 'Supporting (lay) statements or other evidence and hit 'Continue' button

**Expected**
Upload page displays

**Actual**
Upload page displays with BDD SHA alert even though this is not a BDD claim flow

7. Continue the application process until you successfully submit and get to the confirmation page `/disability/file-disability-claim-form-21-526ez/confirmation`

**Expected**
Info or warning alert displays for the status of the claim

**Actual**
Info or warning alert displays for the status of the claim
BDD SHA alert displays

## Acceptance Criteria
- [x] BDD SHA alerts are removed from additional evidence upload view and confirmation pages
- [x] Unit tests pass
- [x] E2E tests cover current code and regression
- [x] End-to-end tests show 0 violations.

### PR DESCRIPTION
**Pull request by christinec-fftc**
## Summary
New BDD Separation Health Assessment alerts are incorrectly showing on the evidence types page and confirmation page (see linked ticket for steps to reproduce)

_(Which team do you work for, does your team own the maintenance of this component?)_
Disability Experience Team

## Related issue(s)
[#58531](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/58531)

## Testing done
 _Describe the steps required to verify your changes are working as expected_
See steps to reproduce, expected vs actual, and url's in linked ticket

_Describe the tests completed and the results_
Unit test fixed, e2e successful, manual test successful

## Screenshots
This type of warning alert should no longer display on the evidence types or confirmation pages when user is filing a non BDD claim
image

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
[n/a] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
[n/a] Documentation has been updated ([link to documentation](#) \*if necessary)
[n/a] Screenshot of the developed feature is added
[n/a] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
[n/a] Events are being sent to the appropriate logging solution
[n/a] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

[n/a] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/tests/pages/additionalDocuments.unit.spec.jsx [hunk 1]
```diff
@@ -127,6 +128,14 @@ describe('526EZ document upload', () => {
...
     }));
 
+    // mock BDD
+    sessionStorage.setItem(
+      SAVED_SEPARATION_DATE,
+      moment()
+        .add(90, 'days')
+        .format('YYYY-MM-DD'),
+    );
+
```
Comment (mchae-nava) [inline src/applications/disability-benefits/all-claims/tests/pages/additionalDocuments.unit.spec.jsx:138]: Nice! Kinda curious what was done in these tests before this, or if were testing around separation date at all. Might be worth a mental note or follow-up item to address as tech debt

**Review by mchae-nava — APPROVED** [review]
Code LGTM, didn't get a chance to reconfirm clicks or accessibility, but will do so later if no one else gets to it

Lots of linting errors that are both OOS and not our problem (yet). I am kinda questioning if we should make an effort to address them at some point, especially the props validation ones, which suggest some intention to enforce a standard

EDIT: Nvm! Looks like you got to it while I was looking

File: src/applications/disability-benefits/all-claims/tests/pages/additionalDocuments.unit.spec.jsx (hunk 1 above)
Comment (christinec-fftc) [reply src/applications/disability-benefits/all-claims/tests/pages/additionalDocuments.unit.spec.jsx:138]: This view didn't have anything specific to BDD flow before and this seemed the easiest option to set up to test this view. The logic to trigger BDD can be found [here](https://github.com/department-of-veterans-affairs/vets-website/blob/58531-remove-alerts/src/applications/disability-benefits/all-claims/utils/index.jsx#L367).

**Comment by christinec-fftc** [comment]
Thanks @mchae-nava. I tried to clean up what i could. The remaining warnings are non-trivial.

File: src/applications/disability-benefits/all-claims/tests/pages/additionalDocuments.unit.spec.jsx (hunk 1 above)
Comment (mchae-nava) [reply src/applications/disability-benefits/all-claims/tests/pages/additionalDocuments.unit.spec.jsx:138]: Gotcha, and sorry, I realize the thought I wrote out was kinda incoherent. Thanks for the context!

**Comment by mchae-nava** [comment]
Clickthrough and automated tests look good!

File: src/applications/disability-benefits/all-claims/components/ConfirmationPoll.jsx [hunk 2]
```diff
@@ -143,8 +137,29 @@ function mapStateToProps(state) {
     disabilities: selectAllDisabilityNames(state),
     submittedAt: state.form.submission.timestamp,
     jobId: state.form.submission.response?.attributes?.jobId,
-    isSubmittingBDD: isBDD(state.form.data) || true,
+    isSubmittingBDD: isBDD(state.form.data) || false,
```
Comment (Mottie) [inline src/applications/disability-benefits/all-claims/components/ConfirmationPoll.jsx:140]: Derp 🙄

**Review by rsmithadhoc — APPROVED** [review]

**Review by adamwhitlock1 — APPROVED** [review]
lgtm

**Review by Mottie — APPROVED** [review]

**Pull request merged**